examples/web_api.rs

```diff
@@ -10,80 +10,91 @@ use hyper::{Body, Chunk, Client, Method, Request, Response, Server, StatusCode,
 use hyper::client::HttpConnector;
 use hyper::service::service_fn;
 
-#[allow(unused, deprecated)]
-use std::ascii::AsciiExt;
-
 static NOTFOUND: &[u8] = b"Not Found";
-static URL: &str = "http://127.0.0.1:1337/web_api";
+static URL: &str = "http://127.0.0.1:1337/json_api";
 static INDEX: &[u8] = b"<a href=\"test.html\">test.html</a>";
-static LOWERCASE: &[u8] = b"i am a lower case string";
+static POST_DATA: &str = r#"{"original": "data"}"#;
+
+type ResponseFuture = Box<Future<Item=Response<Body>, Error=hyper::Error> + Send>;
+
+fn client_request_response(client: &Client<HttpConnector>) -> ResponseFuture {
+     let req = Request::builder()
+         .method(Method::POST)
+         .uri(URL)
+         .header(header::CONTENT_TYPE, "application/json")
+         .body(POST_DATA.into())
+         .unwrap();
+
+     Box::new(client.request(req).map(|web_res| {
+         // Compare the JSON we sent (before) with what we received (after):
+         let body = Body::wrap_stream(web_res.into_body().map(|b| {
+             Chunk::from(format!("<b>POST request body</b>: {}<br><b>Response</b>: {}",
+                                 POST_DATA,
+                                 std::str::from_utf8(&b).unwrap()))
+         }));
+
+         Response::new(body)
+     }))
+}
+
+fn api_post_response(req: Request<Body>) -> ResponseFuture {
+    // A web api to run against
+    Box::new(req.into_body()
+        // A body has multiple chunks, have to combine them before deserializing:
+        .fold(Vec::new(), |mut accumulator, chunk| {
+            accumulator.extend(&chunk[..]);
+            future::ok::<_, hyper::Error>(accumulator)
```

Comment:
@seanmonstar Feel free to review this closer. I'm new to both hyper and rust. Is there a nicer way to convert the body chunks to a vector / string for use with `serde_json::from_str`?

Reply:
You can just use `into_body().concat2()` instead of the `fold`.

Reply:
Thanks, fixed!